### PR TITLE
fix: add linting step and fix linting issues

### PR DESCRIPTION
**Pull request by osamamagdy**

### CODE REVIEW

**Comment by osamamagdy** [comment]
@ankitm123 could you review this?

File: .lighthouse/jenkins-x/pullrequest.yaml [hunk 1]
```diff
@@ -26,11 +26,13 @@ spec:
           resources: {}
         - name: build-make-linux
           resources: {}
+        - name: make-lint
```
Comment (ankitm123) [inline .lighthouse/jenkins-x/pullrequest.yaml]: add a separate file `lint.yaml`, see: https://github.com/jenkins-x-plugins/jx-secret/blob/main/.lighthouse/jenkins-x/lint.yaml, also check the triggers file.

File: go.mod [hunk 2]
```diff
@@ -15,6 +15,7 @@ require (
 	github.com/spf13/cobra v1.2.0
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
+	golang.org/x/tools v0.1.10 // indirect
```
Comment (ankitm123) [inline go.mod:22]: why this change?

File: pkg/cmd/ingress/ingress.go [hunk 3]
```diff
@@ -120,217 +117,6 @@ func (o *Options) Run() error {
 	return requirementsResource.SaveConfig(requirementsFileName)
 }
 
-func (o *Options) discoverIngressDomain(requirements *jxcore.RequirementsConfig, requirementsFileName string) error {
```
Comment (ankitm123) [inline pkg/cmd/ingress/ingress.go:123]: unused functions?

File: cmd/docs/man_docs.go [hunk 4]
```diff
@@ -123,25 +123,38 @@ func fillHeader(header *GenManHeader, name string) error {
...
-%% %s
-%% %s
-# NAME
-`, header.Title, header.Section, header.Source, header.Manual))
-	buf.WriteString(fmt.Sprintf("%s \\- %s\n\n", dashedName, cmd.Short))
-	buf.WriteString("# SYNOPSIS\n")
-	buf.WriteString(fmt.Sprintf("**%s**\n\n", cmd.UseLine()))
-	buf.WriteString("# DESCRIPTION\n")
-	buf.WriteString(description + "\n\n")
+	if _, err := buf.WriteString(fmt.Sprintf(`%% %s(%s)
```
Comment (ankitm123) [inline cmd/docs/man_docs.go:132]: does `make docs` work after this change?

**Review by ankitm123 — COMMENTED** [review]
Few minor comments.
remember to run these commands locally:
* go mod tidy
* make build
* make lint
* make test
* make docs

File: go.mod (hunk 2 above)
Comment (osamamagdy) [reply go.mod:22]: It was auto-generated by one of the commands

File: .lighthouse/jenkins-x/pullrequest.yaml (hunk 1 above)
Comment (osamamagdy) [reply .lighthouse/jenkins-x/pullrequest.yaml]: Done, pushing them now

File: pkg/cmd/ingress/ingress.go (hunk 3 above)
Comment (osamamagdy) [reply pkg/cmd/ingress/ingress.go:123]: yes, this is what the linter said

File: cmd/docs/man_docs.go (hunk 4 above)
Comment (osamamagdy) [reply cmd/docs/man_docs.go:132]: yes, it is running but gave me warnings about some files that do not exist, I don't think those warnings are relevant to this change.
I didn't run make docs before, but it is working now

**Comment by osamamagdy** [comment]
@ankitm123 when running make lint again after `go mod tidy`, which I didn't run the first time, it gave me more linting issues that didn't appear before and I didn't manage to solve these
```
pkg/cmd/job/job_test.go:3: File is not `goimports`-ed (goimports)
import (
```
```
pkg/cmd/install/install.go:168:2: rangeValCopy: each iteration copies 1000 bytes (consider pointers or indexing) (gocritic)
        for _, p := range podList.Items {
        ^
```

```
pkg/cmd/pods/pods.go:6: File is not `goimports`-ed (goimports)
```

```
pkg/cmd/pods/pods_test.go:4: File is not `goimports`-ed (goimports)
        "context"
```

**Comment by ankitm123** [comment]
So for goimports issue, I recommend installing the go plugin in vscode, and making it run goimport on save (There are resources u can check) -  but all it does is gofmt with checking the order of the imports.

As for rangeValCopy issue from gocritic, you just need to index the value inside the loop. So something like this:
```
for k := range podList.Items {
 p := podList.Items[k]
 ....
```

**Comment by ankitm123** [comment]
Also, please squash all the changes into a single commit before marking it ready for review. https://jenkins-x.io/community/code/#squash-and-rebase

**Comment by osamamagdy** [comment]
@ankitm123 Done

File: .lighthouse/jenkins-x/lint.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        steps:
+        - image: uses:jenkins-x/jx3-pipeline-catalog/tasks/git-clone/git-clone-pr.yaml@versionStream
+          name: ""
+          resources: {}
+        - name: make-lint
+          resources: {}
+  podTemplate: {}
+  serviceAccountName: tekton-bot
+  timeout: 30m0s
+status: {}
```
Comment (ankitm123) [inline .lighthouse/jenkins-x/lint.yaml]: NIT: add new line

File: .lighthouse/jenkins-x/pullrequest.yaml (hunk 1 above)
Comment (ankitm123) [reply .lighthouse/jenkins-x/pullrequest.yaml]: remove the step from this file, as we already have a separate lint.yaml file.

File: .lighthouse/jenkins-x/release.yaml [hunk 6]
```diff
@@ -39,5 +39,5 @@ spec:
           resources: {}
   podTemplate: {}
   serviceAccountName: tekton-bot
-  timeout: 240h0m0s
+  timeout: 2h0m0s
```
Comment (ankitm123) [inline .lighthouse/jenkins-x/release.yaml:42]: 💯

**Review by ankitm123 — COMMENTED** [review]
few minor comments, otherwise looks good!

File: .lighthouse/jenkins-x/triggers.yaml [hunk 7]
```diff
@@ -6,13 +6,20 @@ spec:
...
+    rerun_command: /test lint
+    source: "lint.yaml"
   postsubmits:
   - name: release
     context: "release"
     source: "release.yaml"
     branches:
     - ^main$
-    - ^master$
+    - ^master$
```
Comment (ankitm123) [inline .lighthouse/jenkins-x/triggers.yaml]: NIT: missing new line.

File: .golangci.yaml [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+run:
+  timeout: 30m
+  skip-dirs:
+    - cmd/docs
+# golangci.com configuration
+# https://github.com/golangci/golangci/wiki/Configuration
+service:
+  golangci-lint-version: 1.29.x # use the fixed version to not introduce new linters unexpectedly
+  prepare:
+    - echo "here I can run custom commands, but no preparation needed for this repo"
```
Comment (ankitm123) [inline .golangci.yaml]: NIT: new line
